### PR TITLE
Kernels

### PR DESCRIPTION
**Pull request by goastler**
Kernels work from the workshop

I have moved all of the work into the kernel transformer file. I've then moved this file into contrib as it's very rough around the edges...

### CODE REVIEW

File: playground.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import numpy as np
```
Comment (mloning) [inline playground.py]: please remove the playground file from the PR ;-)

File: sktime/transformers/pandas_to_numpy.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from sktime.transformers.base import BaseTransformer
+
+from sktime.utils.transformations import tabularise
+import pandas as pd
+
+class PandasToNumpy(BaseTransformer):
```
Comment (mloning) [inline sktime/transformers/pandas_to_numpy.py]: Do we need a transformer for this one? Transformers should return dataframes to be compatible in pipelines, if you really need a transformer, I'd suggest to make it a helper class (possibly hidden) and to put it into the utils submodule

File: sktime/classifiers/proximity.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Proximity Forest: An effective and scalable distance-based classifier for time series
```
Comment (mloning) [inline sktime/classifiers/proximity.py]: why is prox forest in this PR too?

File: sktime/distances/extra_things.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#if __name__ == "__main__":
```
Comment (mloning) [inline sktime/distances/extra_things.py]: please rename file or remove if not necessary

File: sktime/utils/parallel.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from joblib import Parallel
+
+def generate_dataset_batches(X, parallel):
```
Comment (mloning) [inline sktime/utils/parallel.py]: what is this for? some short docstring description would be helpful

File: train_svm.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import sklearn.svm
```
Comment (mloning) [inline train_svm.py]: please turn into a test file or remove

File: playground.py (hunk 1 above)
Comment (goastler) [reply playground.py]: done

File: sktime/transformers/pandas_to_numpy.py (hunk 2 above)
Comment (goastler) [reply sktime/transformers/pandas_to_numpy.py]: > * mixed with prox forest additions
That's residual from the workshop, these two became intertwined due to distance measures. Once PF PR has been approved I'll merge dev into kernels and it should straighten out.
> * requires some cleaning/reorganisation before merging
Yes, I've maintained this is the case for a while. It's a sizeable job which I'm not prepared to do right now, hence why it's going into contrib (at least that was the idea). I'll sort it out as and when but not sure whether the PR should be approved to get it into contrib (as @TonyBagnall wanted) or to wait until it's polished
> * if you still want to add it now, please move files into contrib submodule
I've now moved everything to do with kernels into the contrib folder

Comment (goastler) [reply sktime/transformers/pandas_to_numpy.py]: I don't know tbh, kernels won't work with dataframes...

File: sktime/classifiers/proximity.py (hunk 3 above)
Comment (goastler) [reply sktime/classifiers/proximity.py]: yes, see above

File: sktime/distances/extra_things.py (hunk 4 above)
Comment (goastler) [reply sktime/distances/extra_things.py]: done

File: sktime/utils/parallel.py (hunk 5 above)
Comment (goastler) [reply sktime/utils/parallel.py]: Legacy parallel code, will delete it

File: train_svm.py (hunk 6 above)
Comment (goastler) [reply train_svm.py]: removed

**Pull request closed without merging**